arch/arm/core/aarch32/Kconfig

```diff
@@ -215,6 +215,24 @@ config ARM_NONSECURE_FIRMWARE
 	  resources of the Cortex-M MCU, and, therefore, it shall avoid
 	  accessing them.
 
+config ARM_STORE_EXC_RETURN
+	bool
+	default y if FPU_SHARING || ARM_NONSECURE_FIRMWARE
```

Comment:
@oyvindronningstad do we need to change the `ARM_NONSECURE_FIRMWARE `to `ARM_NONSECURE_PREEMTIBLE_SECURE_CALLS` in the dependencies here?

In other words, when we do not support preemptible calls, and we do not have FPU, do we need to store the EXC_RETURN?

Reply:
Yes, we need it to correctly keep the S/ES/DCRS flags.

Reply:
I am not sure we need to do anything if we do not switch security domains at swap. If we only switch from NS threads to NS threads, I mean, why do we need to store and restore the EXC_RETURN?